### ISSUE TITLE
Nix Piston: extra build commands

### ISSUE THREAD

**Issue by Brikaa** [engineer-man/piston#450]
How do we handle extra commands that are to be executed after building a Nix package in Nix Piston? Like the ones that are used to install Pip and NuGet packages. I see a `packages` attribute that `piston.mkRuntime` takes and returns, is that intended to be used for extra commands?

**Comment by HexF** [comment]
That is ment to be for a nix package set as nix can handle installing packages from many different languages. 
It does nothing now, but I plan to enable it for some packages so we can get dynamic package installs.

**Comment by Brikaa** [comment]
Okay so what about extra build commands? Like installing PIP and NuGet packages?

**Comment by HexF** [comment]
yeah but I haven't thought that far ahead yet - afaik python packages work fine when installed with nix, same with Ruby and haskell

**Comment by Brikaa** [comment]
Got it. Actually now I see the suggested extra commands attribute wouldn't be of much use.

**Comment by Brikaa** [comment]
Some runtimes need this, so I implemented it in the pull request referenced above

**Comment by Brikaa** [comment]
Turns out, most of the time, Nix provides ways achieve this.
